Mage.Sets/src/mage/cards/g/GiadaFontOfHope.java

```diff
@@ -0,0 +1,115 @@
+package mage.cards.g;
+
+import mage.MageInt;
+import mage.Mana;
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldControlledTriggeredAbility;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.VigilanceAbility;
+import mage.abilities.mana.ConditionalColoredManaAbility;
+import mage.abilities.mana.conditional.ConditionalSpellManaBuilder;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.filter.FilterPermanent;
+import mage.filter.FilterSpell;
+import mage.filter.common.FilterControlledCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+
+import java.util.UUID;
+
+/**
+ * @author woshikie
+ */
+public final class GiadaFontOfHope extends CardImpl {
+
+    private static final FilterPermanent OTHER_ANGEL_YOU_CONTROL_FILTER = new FilterControlledCreaturePermanent(SubType.ANGEL, "other Angel you control");
+    private static final FilterSpell ANGEL_SPELL_FILTER = new FilterSpell("an Angel spell");
+
+    static {
+        OTHER_ANGEL_YOU_CONTROL_FILTER.add(AnotherPredicate.instance);
+        ANGEL_SPELL_FILTER.add(SubType.ANGEL.getPredicate());
+    }
+
+    public GiadaFontOfHope(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{W}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.ANGEL);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Each other Angel you control enters the battlefield with an additional +1/+1 counter on it for each Angel you already control.
+        this.addAbility(
+                new EntersBattlefieldControlledTriggeredAbility(
```

Comment:
I was referencing code from [[Emiel The Blessed]], not sure if I should change the Ability and Effect type to the suggested ones?

Reply:
Emiel is a triggered ability, unlike this.  If it's worded like "When/Whenever (something happens)...", that's a trigger.  This card doesn't have that wording.  So if you play a 3/3 angel, it's going to enter with counters.  No opportunity for an opponent to lightning bolt it since there's no trigger to respond to.

Reply:
added a commit to change implementation for enters battlefield ability.
Referenced cards:
- [[Bloodspore Thrinax]]
- [[Tayam, Luminous Enigma]]